test/honest_test.go

```diff
@@ -2,254 +2,170 @@ package test
 
 import (
 	"fmt"
+	"math/rand"
 	"testing"
 
 	"github.com/filecoin-project/go-f3/sim"
 	"github.com/filecoin-project/go-f3/sim/signing"
 	"github.com/stretchr/testify/require"
 )
 
-func TestHonest_ChainAgreement(t *testing.T) {
+func TestHonest_Agreement(t *testing.T) {
 	t.Parallel()
-	tests := []struct {
-		name             string
-		options          []sim.Option
-		participantCount int
-	}{
-		{
-			name:             "sync singleton no signing",
-			options:          syncOptions(),
-			participantCount: 1,
-		},
-		{
-			name:             "sync singleton bls",
-			options:          syncOptions(sim.WithSigningBackend(signing.NewBLSBackend())),
-			participantCount: 1,
-		},
-		{
-			name:             "sync pair no signing",
-			options:          syncOptions(),
-			participantCount: 2,
-		},
-		{
-			name:             "sync pair bls",
-			options:          syncOptions(sim.WithSigningBackend(signing.NewBLSBackend())),
-			participantCount: 2,
-		},
-		{
-			name:             "async pair no signing",
-			options:          asyncOptions(t, 1413),
-			participantCount: 2,
-		},
-		{
-			name:             "async pair bls",
-			options:          asyncOptions(t, 1413, sim.WithSigningBackend(signing.NewBLSBackend())),
-			participantCount: 2,
-		},
+	maxParticipantCases := 20
+	if testing.Short() {
+		// Reduce max participants to 5 during short test for faster completion.
+		maxParticipantCases = 5
 	}
-	for _, test := range tests {
-		test := test
-		t.Run(test.name, func(t *testing.T) {
-			tsg := sim.NewTipSetGenerator(tipSetGeneratorSeed)
-			baseChain := generateECChain(t, tsg)
-			targetChain := baseChain.Extend(tsg.Sample())
-			test.options = append(test.options,
-				sim.WithBaseChain(&baseChain),
-				sim.AddHonestParticipants(test.participantCount, sim.NewFixedECChainGenerator(targetChain), uniformOneStoragePower))
-			sm, err := sim.NewSimulation(test.options...)
-			require.NoError(t, err)
 
-			require.NoErrorf(t, sm.Run(1, maxRounds), "%s", sm.Describe())
-			requireConsensusAtFirstInstance(t, sm, targetChain.Head())
-		})
+	// Tests with BLS are super slow; try fewer participant counts for them.
+	blsParticipantCount := []int{2, 4, 10}
+
+	// The number of honest participants for which every table test is executed.
+	participantCounts := make([]int, maxParticipantCases)
+	for i := range participantCounts {
+		participantCounts[i] = (i + 1) * 3
```

Comment:
This approach looks promising, but for this one I think it's more important to densely cover the low numbers rather than spread out to high ones. E.g. 4 is a really good case, as the first one where a participant can disagree and be overruled by the others. So suggest `3+i` rather than `3*i`.